crates/ruma-api-macros/src/version.rs

```diff
@@ -0,0 +1,51 @@
+use std::{convert::TryInto, num::NonZeroU8, str::FromStr};
+
+use proc_macro2::TokenStream;
+use quote::ToTokens;
+use syn::{parse::Parse, Error, LitFloat};
+
+#[derive(Clone)]
+pub struct MatrixVersionLiteral {
+    maj: NonZeroU8,
+    min: u8,
+}
+
+impl Parse for MatrixVersionLiteral {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let fl: LitFloat = input.parse()?;
+
+        if !fl.suffix().is_empty() {
+            return Err(Error::new_spanned(
+                fl,
+                "matrix version variable contained invalid float suffix",
```

Comment:
The fact this is a float literal for Rust's lexer is an implementation detail, can we have an error message more like this:

```suggestion
                "matrix version has to be two number separated by `.`",
```